index.js

```diff
@@ -23,77 +20,57 @@ async function run() {
         owner: owner,
         repo: repo,
       })
+
       if (latestRelease) {
         baseRef = latestRelease.data.tag_name
       } else {
-        core.setFailed(
-          `There are no releases on ${owner}/${repo}. Tags are not releases.`,
-        )
+        core.setFailed(`There are no releases on ${owner}/${repo}. Tags are not releases.`)
       }
     }
 
-    console.log(`head-ref: ${headRef}`)
-    console.log(`base-ref: ${baseRef}`)
+    if (!!headRef && !!baseRef && regexp.test(headRef) && regexp.test(baseRef)) {
+      var resp
 
-    if (
-      !!headRef &&
-      !!baseRef &&
-      regexp.test(headRef) &&
-      regexp.test(baseRef)
-    ) {
-      getChangelog(headRef, baseRef, owner + '/' + repo)
-    } else {
-      core.setFailed(
-        'Branch names must contain only numbers, strings, underscores, periods, and dashes.',
-      )
-    }
-  } catch (error) {
-    core.setFailed(error.message)
-  }
-}
-
-async function getChangelog(headRef, baseRef, repoName) {
-  try {
-    let output = ''
-    let err = ''
-
-    // These are option configurations for the @actions/exec lib`
-    const options = {}
-    options.listeners = {
-      stdout: (data) => {
-        output += data.toString()
-      },
-      stderr: (data) => {
-        err += data.toString()
-      },
-    }
-    options.cwd = './'
-
-    await exec.exec(
-      `${src}/changelog.sh`,
-      [headRef, baseRef, repoName],
-      options,
-    )
+      try {
+        resp = await generateReleaseNotes(owner, repo, baseRef, headRef)
+      } catch (err) {
+        core.setFailed(`Could not generate changelog between references because: ${err.message}`)
+        process.exit(1)
+      }
 
-    if (output) {
-      output = jirafyChangelog(output)
       console.log(
         '\x1b[32m%s\x1b[0m',
-        `Changelog between ${baseRef} and ${headRef}:\n${output}`,
+        `Changelog between ${baseRef} and ${headRef}:\n${resp.data.body}`,
       )
-      core.setOutput('changelog', output)
+
+      core.setOutput('changelog', jirafyChangelog(resp.data.body))
+
     } else {
-      core.setFailed(err)
-      process.exit(1)
+      core.setFailed('Branch names must contain only numbers, strings, underscores, periods, and dashes.')
     }
-  } catch (err) {
-    core.setFailed(
-      `Could not generate changelog between references because: ${err.message}`,
-    )
-    process.exit(0)
+  } catch (error) {
+    core.setFailed(error.message)
   }
 }
 
+/**
+ * Generate a name and body describing a release via Github generate-notes API.
+ * @param {*} owner 
+ * @param {*} repo 
+ * @param {*} previousTag 
+ * @param {*} tag 
+ * @returns Object { name, body }
+ */
+async function generateReleaseNotes(owner, repo, previousTag, tag) {  
+  return await octokit.request(`POST /repos/${owner}/${repo}/releases/generate-notes`, {
```

Comment:
Could we use octokit's built-in functions instead of making a custom request?
https://octokit.github.io/rest.js/v18#repos-generate-release-notes

Is this method useful? How much value is there in making our own wrapper that just defers to Octokit, vs. just calling Octokit directly where needed?